### PR TITLE
Feature/debug menu

### PR DESCRIPTION
**Pull request by ThomasSession**
Added a debug menu.

For now it simply allows us to switch between mainnet and testnet.

The debug menu can be accessed in two ways:
- Via an app shortcut (long pressing on the app icon > Debug Menu)
- In the Settings page, under `Debug Menu`

The debug menu should not be accessible in release builds.

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt [hunk 1]
```diff
@@ -498,27 +509,33 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
 
             Cell {
                 Column {
+                    // add the debug menu in non release builds
+                    if (BuildConfig.BUILD_TYPE != "release") {
```
Comment (AL-Session) [inline app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt:513]: Prevent string comparison and any potential typos?
```suggestion
                    if (BuildConfig.BUILD_TYPE == BuildConfig.DEBUG) {
```

File: app/src/main/java/org/thoughtcrime/securesms/debugmenu/DebugMenu.kt [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            }
+        }
+
+        // Alert dialogs
+        if (uiState.showEnvironmentWarningDialog) {
+            AlertDialog(
+                onDismissRequest = { sendCommand(HideEnvironmentWarningDialog) },
+                title = "Are you sure you want to switch environments?",
+                text = "Changing this setting will result in all conversations and Snode data being cleared...",
+                showCloseButton = false, // display the 'x' button
```
Comment (AL-Session) [inline app/src/main/java/org/thoughtcrime/securesms/debugmenu/DebugMenu.kt:70]:
```suggestion
                showCloseButton = false, // don't display the 'x' button
```

**Review by AL-Session — APPROVED** [review]
Looking good! Two suggestions for you to address as you see fit, only.

**Review by simophin — APPROVED** [review]

File: app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt (hunk 1 above)
Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt:513]: @AL-Session I considered this but I want to add a QA build type in the future. That build type should have the debug menu available, so unfortunately it needs to be compared to `release`.

**Pull request merged**